### ISSUE TITLE
Question - Hotfix vs actual develop branch

### ISSUE THREAD

**Issue by rlejeune** [release-drafter/release-drafter#520]
Been using your action for some times now and I really like it. There's one use case though that I cannot get to work, and I was wondering if I am missing a parameter or something.

Right now I'm working on, lets say, version 1.1.0, every now and then I push a PR to develop and me release is updated with every PR since 1.0.0. Perfect.

Now I have to push a hotfix to production, so I create a branch with my production version, fix my bug and push it to prod. Once I am done I create a release v1.0.2, and I publish it. 

If I do that, the next time I push a PR to develop, Release Drafter will update my draft release with every PR since the last release, so I now only have the PR since I created v1.0.2 and lost all the ones before that. 

Is there a way to tell Release Drafter to ignore a certain tag, or to get everything from develop since the last release in develop. Or will I have to do manual manipulation every time I'll have to do a hotfix in prod? I'm guessing you guys already had to do that kind of thing, so I came to ask for your input. 

If this is not the place to ask question, sorry about that. I didn't find any other place. 

Thanks!

**Comment by jetersen** [comment]
At the moment it is not possible, but it seems to be a much requested feature.

**Comment by rlejeune** [comment]
I could probably take a look at it and submit a PR. 

What would be the best way to implements this in your opinion?

**Comment by jetersen** [comment]
With #546 you can freely design any release workflow inside your GitHub action's workflow.

**Comment by rlejeune** [comment]
Morning!

This PR does not seems to fix my problem. I just did some testing with it and it's still wiping the entire release and basing what 's in it since last release, which is the hotfix release. 

Maybe I was not clear enough when I explained my problem. 

I am working on a branch, lets say develop since three weeks, my release drafter has all my PR since I started. Lets say 10 of them. My last release was on version v1.1.0, my draft one is v1.2.0.

This morning I have to make a hotfix on master, and release it to my production servers. Once I am done with my hotfix I create a release manually, v1.1.1, stating what I fixed. 

Then I push another PR to develop to keep working on my next release which is due next week. Once my actions runs, release drafter will wipe the actual release and create a new one with all the PR since the last release, which is one PR since for release drafter the last release is the one I create after I pushed my hotfix to master. 

I'd like to have a way to tell release drafter to ignore certain release when it checks the last one on which it'll base the release draft. 

As of now, the code is always getting the last release base on tag/or date, which is what causing me problems.
```
const sortReleases = (releases) => {
  // For semver, we find the greatest release number
  // For non-semver, we use the most recently merged
  try {
    return releases.sort((r1, r2) => compareVersions(r1.tag_name, r2.tag_name))
  } catch (error) {
    return releases.sort(
      (r1, r2) => new Date(r1.created_at) - new Date(r2.created_at)
    )
  }
}
```

So the code will check the latest release, and it'll find v1.1.1. So it'll take redo my draft with every PR since v1.1.1 instead of v1.1.0.

Thank you in advance!

**Comment by SebastianThiebaud** [comment]
@rlejeune Experiencing the same issue even with a simpler git flow (no `develop` branch, only `master`). It seems like `release-drafter` doesn't support our workflow. Since it's based on the latest tag, `hotfix` branches are a no go. Also, in the `git-flow` world, we open release and hotfix branches, do our little magic in these branches, then tag once the release/hotfix is finished. A lot can happen in `master/develop` while this branch is alive.

**Comment by jetersen** [comment]
You would need multiple github action workflow to achieve this and some changes to release drafter.

I think for simplicity you shouldn't run drafter on anything but the development branch.

For hotfixes we could improve it slightly with the following workflow:
```
name: Release Drafter

on:
  push:
    tags:
      - v*
```

And than using the tag to figure out the closest semver ie. a tag with v1.1.1 would find v1.1.0.

Though for more complex git flows.

I think we would actually need to use git cli to figure out common ancestor ie. `git merge-base v1.1.1 v1.1.0`

**Comment by jetersen** [comment]
What about an option to ignore releases?

Perhaps as part of config or as part of the release body?
You could have this inside your release body:
```
<!-- release drafter ignore -->
```
For when you need the draft to be between `1.1.0` and `1.2.0` where a hotfix `1.1.1` exist between.

**Comment by SebastianThiebaud** [comment]
@jetersen I need to think more about all of this. This is definitely tricky and everyone has a slight different git-flow use. What we ask here is to have `release-drafter` being really smart, maybe too smart. Here is the workflow I'm working with if you're curious https://reallifeprogramming.com/git-process-that-works-say-no-to-gitflow-50bf2038ccf7

**Comment by jetersen** [comment]
Now that I think about it you can via the input actually create an action workflow that can provide the necessary input such as common ancestor to use for looking up PRs.

**Comment by rlejeune** [comment]
Would it be possible for release drafter to see on which branch a release has been created? And if it's a different branch than the one on which the action is running then it just ignores it? So you have a draft release that is on develop, and you create a hotfix release on another branch. When release drafter runs it just ignore that release since it has been created on another branch that the one it runs on right now. That way it'd get all the PRs that has been push to lets say develop since last release on develop. 

But, for me, an ignore tag would do the trick if it's easier to implements on your side. I'm ok if the hotfix part has some manual stuff in it since it's not happening that much, well most of the time ;)

**Comment by jetersen** [comment]
GitHub seems to be aware of which branch a release is created on according to this: https://github.com/goharbor/harbor/releases
You can see it is clearly aware of this.
Perhaps the data is there on GitHub.

**Comment by mawkee** [comment]
I have a very similar situation to this. The flow is almost identical to the one @rlejeune described. Since it's on an app and we have a particular workflow, continuous delivery is not very practical for our usage, so hotfixes on the "production" branch happens from time to time.

I think the `<!-- release drafter ignore -->` approach seems very sensible. It requires some manual work, but that's to be expected on a hotfix. And leaving something on the config file to deal with this would probably be more trouble than it's worth.

Today I'm manually copying & pasting the contents of the latest release before I do a hotfix, and then I manually "merge" the data when I'm ready for the final release.

**Comment by valentijnscholten** [comment]
Just adding here that we're using git flow and currently seem a mismatch between functionality and the git flow branching / release model. Thinking out loud here what you really want I think is that for the dev branch you want the drafter to select PRs that are not yet in master and not yet releases on master.
Then if you create a release branch (from dev), the release drafter would need to look at that release branch and select all PRs that are not yet in master and not yet release in master in the same way.
In parallel work goes on on dev and there the drafter would still need to select any new PRs that are not in master.
Hotfixes in the same way.
So that would mean 3 workflows? And an extra param to which branch the current branch should be compared? (compare to master by default?).

**Comment by mkurz** [comment]
Hi everyone! I think some you face problems you could solve with the `filter-by-commitish` config. I wrote a little guide how to make things work, specially how to see and update the `target_commitish` value of a GitHub release (found at the end of my post): https://github.com/release-drafter/release-drafter/issues/656#issuecomment-806184887 - maybe that helps someone!

**Comment by jetersen** [comment]
In can get there with `target_commitsh` and multiple config files.

Though I would suggest git flow is not the best approach to releases and CI/CD:

> Gitflow is a legacy Git workflow that was originally a disruptive and novel strategy for managing Git branches.
> Gitflow has fallen in popularity in favor of [trunk-based workflows](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development), which are now considered best practices for modern continuous software development and [DevOps](https://www.atlassian.com/devops/what-is-devops) practices.
> Gitflow also can be challenging to use with [CI/CD](https://www.atlassian.com/continuous-delivery). This post details Gitflow for historical purposes.
> https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow
